cdap-elastic/src/main/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -277,8 +271,34 @@ public VersionInfo getVersionInfo() throws IOException {
     }
   }
 
+  /**
+   * Returns a {@link RestHighLevelClient} for talking to elastic search.
+   */
+  private RestHighLevelClient getClient() {
+    RestHighLevelClient client = this.client;
```

Comment:
it looks like we do not need to maintain a local variable here, we can just return this.client after assignment.

Reply:
It is not a safe operation if it is not done inside sync block.
```
if (field != null) {
  // field may set to null by some other thread.
  return field;
}
```